examples/custom-state-storage/mongo-state-storage.js

```diff
@@ -0,0 +1,62 @@
+const mongodb = require('mongodb');
+const MongoClient = mongodb.MongoClient;
+const Bluebird = require('bluebird');
+
+Bluebird.promisifyAll(MongoClient);
+
+class MongoDbStore {
+    constructor() {
+        this.url = 'mongodb://localhost/test';
+    }
+
+    connect() {
+        return MongoClient.connect(this.url)
+            .then(client => {
+                return client.db()
+            })
+    }
+
+    load(fn) {
+        return this.connect()
+            .then(db => db.collection('db_migrations').find().toArray())
+            .then(data => {
+                if (!data.length) return fn(null, {});
+                const store = data[0];
+                if (!MongoDbStore.hasProperty(store, 'lastRun') ||
+                    !MongoDbStore.hasProperty(store, 'migrations')) {
+                    return fn(new Error('I`nvalid store file'))
+                }
+                return fn(null, store)
+            }).catch(fn)
+    }
+
+    static hasProperty(store, property) {
+        return Object
+            .prototype
+            .hasOwnProperty
+            .call(store, property);
+    }
+
+    save(set, fn) {
+        return this.connect()
+            .then(db => db.collection('db_migrations')
+                .update({},
+                    {
+                        $set: {
+                            lastRun: set.lastRun,
+                        },
+                        $push: {
+                            migrations: {$each: set.migrations},
+                        },
+                    },
+                    {
+                        upsert: true,
+                        multi: true,
+                    }
+                ))
+            .then(result => fn(null, result))
+            .catch(fn)
+    }
+}
+
+module.exports = MongoDbStore;
```

Comment:
To be consistent with the `pg` example, this should not be exported but used to run migrations.  See: https://github.com/tj/node-migrate/blob/master/examples/custom-state-storage/custom-state-storage.js#L51-L66

Reply:
If I use migrate.load() and pass custom state store it still fails with `TypeError: Store is not a constructor` but it works fine with `module.exports = MongoDbStore;`.

Reply:
I just meant that it should show how to use the state store like that other example does.  The goal of the examples is not that they are usable outside the package, just that they illustrate how one might do it.  Basically I was thinking just this:

```javascript
migrate.load({
  stateStore: MongoDbStore
}, function (err, set) {
  if (err) {
    throw err
  }

  set.up((err2) => {
    if (err2) {
      throw err2
    }

    console.log('Migrations successfully ran')
  })
})
```